python_modules/dagster/dagster/_core/pipes/client.py

```diff
@@ -54,29 +54,51 @@ def run(
 
 @experimental
 class PipesClientCompletedInvocation:
-    def __init__(self, results: Sequence["PipesExecutionResult"]):
-        self._results = results
+    def __init__(
+        self,
+        session: PipesSession,
+    ):
+        self._session = session
 
-    def get_results(self) -> Sequence["PipesExecutionResult"]:
+    def get_results(
+        self,
+        *,
+        implicit_materializations=True,
+    ) -> Sequence["PipesExecutionResult"]:
```

Comment:
i went pretty far on an impl that exposed separate methods with "reported" in the name to distinguish but came back to the ol bool toggle approach. Open to any other ideas or names